### DISCUSSION TITLE
Prevent Text From Truncating

### DISCUSSION THREAD

**Discussion by Tyler-Lopez** [patrykandpatrick/vico#953] [Questions]
### Question

Hi there,

I would like to know if it is possible to prevent text from truncating into `...`.

This is not asking for there to be additional padding where `...` happens, but instead to just prevent that behavior entirely.

If I find the solution I will include it in this thread.

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @Tyler-Lopez. Can you give more context to the question? Is it about `HorizontalAxis`?

**Reply by Tyler-Lopez** [reply]
Hi @Gowsky! Yes, it is. Specifically, I would like to prevent the text, when there isn't enough room to show it from being "...". 

I'd like to have much more freedom here.

For example, I would like some of these possibilities - and I note these possibilities could already exist in custom implementations:

* Simply allow text to clip when there isn't enough room rather than truncate it - this is especially useful as sometimes the decision to not clip seems a touch overly cautious.
* Instead of text clipping as "...", simply have it not render at all.

**Reply by Gowsky** [reply]
> Simply allow text to clip when there isn't enough room rather than truncate it - this is especially useful as sometimes the decision to not clip seems a touch overly cautious.

If you're seeing excessive truncation, it's most likely a result of horizontal padding applied to the axis labels, so you can resolve it by removing the padding. (Some horizontal padding is applied by default.) `TextComponent` uses `StaticLayout`, just like `TextView`, so truncation should generally work as expected.

Is there any other case in which you need to allow overlaps? It would seem that they're generally undesirable.

> Instead of text clipping as "...", simply have it not render at all.

Could you share some extra context? In what case are you observing truncation to just "..."? With the default settings, for example, this shouldn't happen, since `AxisItemPlacer.Horizontal.aligned` adjusts the label spacing so that the labels fit without truncation.
